code/crates/core-consensus/src/handle/driver.rs

```diff
@@ -194,36 +194,39 @@ where
                 "Proposing value"
             );
 
-            let signed_proposal = sign_proposal(co, proposal).await?;
-
-            if signed_proposal.pol_round().is_defined() {
-                perform!(
-                    co,
-                    Effect::RestreamValue(
-                        signed_proposal.height(),
-                        signed_proposal.round(),
-                        signed_proposal.pol_round(),
-                        signed_proposal.validator_address().clone(),
-                        signed_proposal.value().id(),
-                        Default::default()
-                    )
-                );
+            // Only sign and publish if we're in the validator set
+            if state.is_validator() {
```

Comment:
Not sure about the proposal filtering, seems we are trying to defend against a bug where `get_proposer()` returns a non-validator/ full node address...but in this case the `is_validator()` should also fail.

Reply:
Yes it's just defensive coding. Maybe we can change the check to do a `debug_assert` or even an `assert` in order to ensure this code path is never hit.